tabular/src/autogluon/tabular/testing/fit_helper.py

```diff
@@ -0,0 +1,456 @@
+            "problem_type": MULTICLASS,
+            "label": "Cover_Type",
+        },
+        # Regression with mixed feature-types, skewed Y-values.
+        ames={
+            "url": "https://autogluon.s3.amazonaws.com/datasets/AmesHousingPriceRegression.zip",
+            "name": "AmesHousingPriceRegression",
+            "problem_type": REGRESSION,
+            "label": "SalePrice",
+        },
+        # Regression with multiple text field and categorical
+        sts={
+            "url": "https://autogluon-text.s3.amazonaws.com/glue_sts.zip",
+            "name": "glue_sts",
+            "problem_type": REGRESSION,
+            "label": "score",
+        },
+    )
+
+    toy_map = dict(
+        toy_binary=generate_toy_binary_dataset,
+        toy_multiclass=generate_toy_multiclass_dataset,
+        toy_regression=generate_toy_regression_dataset,
+        toy_quantile=generate_toy_quantile_dataset,
+        toy_binary_10=generate_toy_binary_10_dataset,
+        toy_multiclass_10=generate_toy_multiclass_10_dataset,
+        toy_regression_10=generate_toy_regression_10_dataset,
+        toy_quantile_10=generate_toy_quantile_10_dataset,
+        toy_multiclass_30=generate_toy_multiclass_30_dataset,
+    )
+
+    @staticmethod
+    def load_dataset_toy(name: str) -> tuple[pd.DataFrame, pd.DataFrame, dict]:
+        return DatasetLoaderHelper.toy_map[name]()
+
+    @staticmethod
+    def load_dataset(name: str, directory_prefix: str = "./datasets/") -> tuple[pd.DataFrame, pd.DataFrame, dict]:
+        if name in DatasetLoaderHelper.toy_map:
+            return DatasetLoaderHelper.load_dataset_toy(name=name)
+        dataset_info = copy.deepcopy(DatasetLoaderHelper.dataset_info_dict[name])
+        train_file = dataset_info.pop("train_file", "train_data.csv")
+        test_file = dataset_info.pop("test_file", "test_data.csv")
+        name_inner = dataset_info.pop("name")
+        url = dataset_info.pop("url", None)
+        train_data, test_data = DatasetLoaderHelper.load_data(
+            directory_prefix=directory_prefix,
+            train_file=train_file,
+            test_file=test_file,
+            name=name_inner,
+            url=url,
+        )
+
+        return train_data, test_data, dataset_info
+
+    @staticmethod
+    def load_data(
+        directory_prefix: str,
+        train_file: str,
+        test_file: str,
+        name: str,
+        url: str | None = None,
+    ):
+        if not os.path.exists(directory_prefix):
+            os.mkdir(directory_prefix)
+        directory = directory_prefix + name + "/"
+        train_file_path = directory + train_file
+        test_file_path = directory + test_file
+        if (not os.path.exists(train_file_path)) or (not os.path.exists(test_file_path)):
+            # fetch files from s3:
+            print("%s data not found locally, so fetching from %s" % (name, url))
+            zip_name = download(url, directory_prefix)
+            unzip(zip_name, directory_prefix)
```

Comment:
Are files assumed to always be zipped?

Reply:
This is based on the old logic from 2019, simply copy pasted out of tests. It should ideally be improved, but this is a refactor for another time.